rstar/src/algorithm/rstar.rs

```diff
@@ -68,13 +71,13 @@ impl InsertionStrategy for RStarInsertionStrategy {
                     match forced_insertion::<T, Params>(root, node_to_reinsert, target_height) {
                         InsertionResult::Split(node) => insertion_stack.push(PerformSplit(node)),
                         InsertionResult::Reinsert(_, _) => {
-                            panic!("Unexpected reinsert. This is a bug in rstar.")
+                            Err("Unexpected reinsert. This is a bug in rstar.")?
```

Comment:
https://rust-lang.github.io/rust-clippy/master/index.html#try_err

Reply:
(Actually, I think running `cargo fmt --all -- --check` and `cargo clippy --all -- --deny warnings` as part of the CI would be nice.)

Reply:
To give a bit more background about _why_ this panic should never be hit: The algorithm will from time to time (when a node overflows) attempt to reinsert (remove and insert again) a point to improve the tree's internal structure. This is what `InsertionResult::Reinsert(_,_)` means - it tells the top level function (the beginning of the recursion) that some point should be reinserted.
Since inserting that point can lead to _another_ reinsertion (and possibly even an endless loop), reinsertion is only attempted once. That's where `forced_insertion` differs - it will never allow re-inserting an element.

I think the best way to get rid of the panic would be to modify the return type of `forced_insert` to a new `ForcedInsertionResult` enum which doesn't contain a reinsert entry. Then this branch would not appear in the first place.